### PR TITLE
Added Category entity with all related dto, service, mapper and controller.

### PR DESCRIPTION
**Pull request by nklimovych**

### CODE REVIEW

File: src/main/resources/db/changelog/changes/08-create-category-table.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+                  name: description
+                  type: varchar(255)
+                  constraints:
+                    nullable: true
+              - column:
+                  name: deleted
+                  type: bit
+                  defaultValueBoolean: false
+                  constraints:
+                    nullable: false
```
Comment (vikkushnir) [inline src/main/resources/db/changelog/changes/08-create-category-table.yaml]:
```suggestion
                    nullable: false
                    
```

File: src/main/java/mate/academy/bookstore/service/impl/CategoryServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    private final CategoryMapper categoryMapper;
+
+    @Autowired
+    public CategoryServiceImpl(
+            CategoryRepository categoryRepository,
+            CategoryMapper categoryMapper
+    ) {
+        this.categoryRepository = categoryRepository;
+        this.categoryMapper = categoryMapper;
+    }
```
Comment (vikkushnir) [inline src/main/java/mate/academy/bookstore/service/impl/CategoryServiceImpl.java]:
```suggestion
@Service
@RequiredArgsConstructor
public class CategoryServiceImpl implements CategoryService {
    private final CategoryRepository categoryRepository;
    private final CategoryMapper categoryMapper;
```

**Review by vikkushnir — APPROVED** [review]
LGTM! Only few minor comments

File: src/main/java/mate/academy/bookstore/service/CategoryService.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    List<CategoryDto> findAll(Pageable pageable);
+
+    CategoryDto getById(Long id);
+
+    CategoryDto save(CategoryDto categoryDto);
+
+    CategoryDto update(Long id, CategoryDto categoryDto);
+
+    void delete(Long id);
+
```
Comment (andrii-hoienko) [inline src/main/java/mate/academy/bookstore/service/CategoryService.java]:
```suggestion
```

File: src/main/java/mate/academy/bookstore/service/impl/CategoryServiceImpl.java (hunk 2 above)
Comment (andrii-hoienko) [reply src/main/java/mate/academy/bookstore/service/impl/CategoryServiceImpl.java]: agree

File: src/main/java/mate/academy/bookstore/service/impl/CategoryServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+        Category category = categoryMapper.toEntity(categoryDto);
+        category.setId(id);
+        Category savedCategory = categoryRepository.save(category);
+        return categoryMapper.toDto(savedCategory);
+    }
+
+    @Override
+    public void delete(Long id) {
+        categoryRepository.deleteById(id);
```
Comment (andrii-hoienko) [inline src/main/java/mate/academy/bookstore/service/impl/CategoryServiceImpl.java:67]: make check before delete as you did in update

File: src/main/resources/db/changelog/changes/08-create-category-table.yaml (hunk 1 above)
Comment (andrii-hoienko) [reply src/main/resources/db/changelog/changes/08-create-category-table.yaml]: Not resolved

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
Nice! left some comments

File: src/main/java/mate/academy/bookstore/exception/GlobalExceptionHandler.java [hunk 5]
```diff
@@ -38,6 +38,11 @@ protected ResponseEntity<Object> handleMethodRepository(DuplicateIsbnException e
         return getResponseEntity(HttpStatus.BAD_REQUEST, ex.getMessage());
     }
 
+    @ExceptionHandler(DuplicateEntityException.class)
+    protected ResponseEntity<Object> handleMethodRepository(DuplicateEntityException ex) {
```
Comment (yevheniisynytsia) [inline src/main/java/mate/academy/bookstore/exception/GlobalExceptionHandler.java:42]:
```suggestion
    protected ResponseEntity<Object> handleDuplicateEntity(DuplicateEntityException ex) {
```

**Review by yevheniisynytsia — APPROVED** [review]

**Pull request merged**